lib/src/main/kotlin/com/swmansion/starknet/data/serializers/TransactionSerializer.kt

```diff
@@ -0,0 +1,87 @@
+package com.swmansion.starknet.data.serializers
+
+import com.swmansion.starknet.data.types.*
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.SerializationException
+import kotlinx.serialization.descriptors.PrimitiveKind
+import kotlinx.serialization.descriptors.PrimitiveSerialDescriptor
+import kotlinx.serialization.encoding.Decoder
+import kotlinx.serialization.encoding.Encoder
+import kotlinx.serialization.json.*
+
+internal object TransactionSerializer : KSerializer<Transaction> {
+    override val descriptor = PrimitiveSerialDescriptor("Transaction", PrimitiveKind.STRING)
+
+    override fun deserialize(decoder: Decoder): Transaction {
+        require(decoder is JsonDecoder)
+        val element = decoder.decodeJsonElement()
+        val jsonElement = element.jsonObject
+        val typeElement = jsonElement.getOrElse("type") { throw SerializationException("Input element does not contain mandatory field 'type'") }
+
+        val type = decoder.json.decodeFromJsonElement(TransactionType.serializer(), typeElement)
+
+        return when (type) {
+            TransactionType.INVOKE -> deserializeInvoke(decoder, element)
+            TransactionType.DECLARE -> deserializeDeclare(decoder, element)
+            TransactionType.DEPLOY_ACCOUNT -> deserializeDeployAccount(decoder, element)
+            else -> throw IllegalArgumentException("Invalid transaction type '${typeElement.jsonPrimitive.content}'")
+        }
```

Comment:
This seems odd. How should we deserialize other tx types, e.g. `L1_HANDLER`?

If I remember, we didn't use custom deserialization for `Transaction` classes altogether, but if that's something that's necessary, I think it should handle all cases.

(tbh I'm not sure why all tests pass 😅)

Reply:
Right, I've added `L1_HANDLER ` and `DEPLOY`.